### DISCUSSION TITLE
Netzero won't "Download Data" any more ??

### DISCUSSION THREAD

**Discussion by chrisj7903** [netzero-labs/netzero#23] [General]
All of a sudden Netzero does nothing when I click on "Download Data ". Usually it would display various send options (like SMS,email ...) for me to choose from but now literally nothing happens after clicking, It's been working fine until this. Otherwise the App seems to be behaving as usual. I can view everything else.
 t's not the Phone OS because Tesla App still downloads. (restarting App/Phone makes no difference) 
Any suggestions appreciated.

**Comment by zigam** [comment]
Thanks for the report. I'd need to investigate this, can you please send an email to feedback@netzeroapp.io?

**Comment by zigam** [comment]
There was an issue with Android downloads due to an Android API upgrade, fixed now.

**Comment by chrisj7903** [comment]
Hi Ziga,

I just tested a download and can confirm it works just like it used to!
Nice work, thanks for the prompt repairs.

Question: I see some comments on Github that suggest you have been adding
Powerwall2 control functionality to NetZero. I still revert to the Tesla
App for just two things:

1) In Winter I sometimes need to charge my batteries (I have two
Powerwall2) and I want to do it during the low tariff time window (middle
of day) . Currently I manually increase Setting > Powerwall to 100% and
when the battery is sufficiently charged I manually revert back to 10%
reserve. Are you able to add similar manual controls? And for the future:
automate this based on user settings for the charge window.

2) I often quickly check battery SOC % and general power flow during the
day (esp. in Winter) and I still prefer the simpler Tesla home screen
graphic for these quick checks. I would really like an option on NetZero to
have a similar simple home screen showing Battery % SOC and current power
flows for Solar, Home, Powerwall, Grid.

Then I could (& would) ditch the Tesla App altogether :-)

Also are you doing this for fun, or hoping to monetise with a Pro version
or similar?
For what it's worth, I would be happy to pay a modest Google Play fee for
extras like above. Or for free if you prefer :-)

best
Chris

**Reply by zigam** [reply]
1) Yes, you can change Powerwall settings, both as a one-off or on schedule. See the Powerwall Automation feature in the app.

2) The Netzero home screen should convey all of that information (and more), but if there are specific improvements you'd like to see, let us know.

There may be a paid version of Netzero in the future, but it's still early in development.

**Comment by chrisj7903** [comment]
Ziga,
I just realised the Powerwall Automation feature already in the App is
exactly what I was asking for ....
doh! sorry.
many thanks for the great App ...
